### ISSUE TITLE
Generating Single Number Testing Metric

### ISSUE THREAD

**Issue by CodyKurpanek** [ameraner/dsen2-cr#13]
Hello @ameraner,

After testing the pre-trained CARL model on the hold-out test dataset in train_val_test_patches.csv (7899 images) with a batch_size of 1, I see that a csv file is generated with each line containing the metrics for a specific batch. How would I achieve a single number metric as seen in table 1 of your paper? Would I just have to calculate the column-wise average?

After plotting the values from that csv file in a histogram I see that in the MAE and RMSE categories, there are a few testing images where the model achieves equal or better results to the ones recorded in the paper, but in general MAE and RMSE seem to fall around .11 and 0.198 as opposed to .0290 and .0366 seen in the paper, could there be a reason for this?

Thank you

**Comment by ameraner** [comment]
Hi @CodyKurpanek , 
thank you for your questions and sorry for the late answer! Indeed, the final metrics were computed from the average of the single outputs. 
As for the difference in metric results, I cannot think of a good reason.. the public dataset is a bit different compared to the one I actually used (as described in the Readme), but I would have not expected such a big difference...

**Comment by ameraner** [comment]
Hi @CodyKurpanek , I looked into this and now I understand the difference in values between the metrics output of the model and the paper: this is due to the scaling I apply to the input values of the S2 bands, defined here  https://github.com/ameraner/dsen2-cr/blob/9d6cdcf9e262920cb971e34fe298874d77e7a269/Code/dsen2cr_main.py#L42 and applied here https://github.com/ameraner/dsen2-cr/blob/9d6cdcf9e262920cb971e34fe298874d77e7a269/Code/tools/dataIO.py#L390 . The original values of the S2 bands are 0-1 reflectances; the dataset has then been scaled with `*10000`, and in the code I apply `/2000` to bring back the range to 0-5 for numerical stability as proposed in the original DSen2 model. The model MAE and RMSE metrics are output in this 0-5 range, hence, to bring it to 0-1 reflectance units, I (manually) applied a `/5` division to the model metric output. With this, your output and the values in the paper match well again :)

**Comment by CodyKurpanek** [comment]
Hi @ameraner , that is great :) thank you for looking into this. I was suspecting an issue with units, as all other metrics were consistent. In addition, for anyone else looking at this issue, units for SAM need to be manually converted from radians to degrees for the code's output to match the results in the paper.
